src/tools/rbd_mirror/image_replayer/BootstrapRequest.cc

```diff
@@ -194,6 +194,10 @@ void BootstrapRequest<I>::handle_get_client(int r) {
     derr << ": failed to retrieve client: " << cpp_strerror(r) << dendl;
     finish(r);
     return;
+  } else if (m_client.state == cls::journal::CLIENT_STATE_DISCONNECTED) {
+    dout(0) << ": client flagged disconnected, re-registering" << dendl;
+    unregister_client();
```

Comment:
So long as this doesn't result in a split-brain, this is probably fine for short term. I think the intermediate term goal would be to set the ```MirrorPeerState``` state back to ```MIRROR_PEER_STATE_SYNCING``` to indicate that a full re-sync is required and re-connect the client. The long term goal would add the smarts to have multiple sync points / incremental syncs upon disconnect.